lib/velum/file_config.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+require "json"
+require "fileutils"
+require "securerandom"
+
+module Velum
+  # Handle IO operations and data serialisation for configuration files under a specific directory.
+  class FileConfig
+    def initialize(base_dir)
+      @base_dir = base_dir
+    end
+
+    # Return true only if the specified file name exists in the configuration base directory.
+    def exists?(file_name)
```

Comment:
I'm not sure why we need to always pass the `file_name` to all instance methods instead of just making it an instance attribute.
```ruby
def initialize
  @secrets_path = Pathname.new("#{ENV.fetch("VELUM_SECRETS_DIR")}/app_secret.json")
end
```
have a look at the `pathname` gem. then you also don't have to reimplement the `exist?` method.
and then you can do all IO operations much easier, e.g
```ruby
unless @secrets_path.exist?
  @secrets_path.open("w") { |f| f.puts JSON.dump(base: SecureRandom.random_bytes(64).unpack('H*').first }
end

return JSON.parse(@secrets_path.read)[:base]
```